src/dc_tools.rs

```diff
@@ -27,14 +27,10 @@ pub unsafe fn dc_strdup(s: *const libc::c_char) -> *mut libc::c_char {
     let ret: *mut libc::c_char;
     if !s.is_null() {
         ret = strdup(s);
-        if ret.is_null() {
-            exit(16i32);
-        }
+        assert!(!ret.is_null());
```

Comment:
WDYT about changing all these asserts to e.g. `assert!(foo.is_some())`?

Reply:
I don't think `is_some` exists for pointers